screens/Playlist.js

```diff
@@ -0,0 +1,284 @@
+  );
+};
+
+const PlaylistScreen = ({ navigation, route }) => {
+  const {
+    playlistID,
+    playlistTitle,
+    playlistDescription,
+    playlistURI,
+  } = route.params;
+  console.log(route.params);
+  navigation.setOptions({ title: playlistTitle });
+  const insets = useSafeArea();
+  const ref = React.useRef(null);
+
+  useScrollToTop(ref);
+
+  const [isLoading, setLoading] = useState(true);
+  const [isError, setError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    getVideos();
+  }, []);
+  async function getVideos() {
+    try {
+      const storedMedia = await getStoredPlaylist(playlistID);
+
+      if (storedMedia) {
+        setData(JSON.parse(storedMedia));
+        setLoading(false);
+      }
+
+      const fetchedVideos = (await collectData(playlistID)) || [];
+
+      setData(fetchedVideos);
+      setLoading(false);
+      storePlaylistData(playlistID, fetchedVideos);
+    } catch (err) {
+      setError(true);
+      setErrorMessage("Make sure you're connected to the internet.");
+      setLoading(false);
+      Amplitude.logEventWithProperties('ERROR loading playlist items', {
+        error: err,
+      });
+    }
+  }
+  if (isLoading) {
+    return (
+      <View style={[styles.container, { flex: 1, paddingTop: insets.top }]}>
+        <Text bold style={styles.headerTitle}>
+          MEDIA
+        </Text>
+        <Spinner />
+      </View>
+    );
+  }
+
+  if (isError) {
+    return (
+      <View style={[styles.container, { flex: 1, paddingTop: insets.top }]}>
+        <Text bold style={styles.headerTitle}>
+          MEDIA
+        </Text>
+        <View
+          style={[
+            styles.container,
+            { alignItems: 'center', justifyContent: 'center', padding: 16 },
+          ]}
+        >
+          <Title style={[{ textAlign: 'center' }]}>
+            {' '}
+            Oh no! There was an error connecting to YouTube 😞
+          </Title>
+          <Text style={[{ textAlign: 'center', color: Colors.darkGray }]}>
+            {errorMessage}
+          </Text>
+          <Button
+            title={'Retry'}
+            style={styles.notesButton}
+            onPress={() => {
+              setError(false);
+              setLoading(true);
+              getVideos();
+            }}
+          />
+        </View>
+      </View>
+    );
+  }
+
+  return (
+    <HeaderHeightContext.Consumer>
+      {(headerHeight) => (
+        <View style={[styles.mainContainer, { paddingTop: headerHeight }]}>
+          <ScrollView>
+            <Image
+              source={{ uri: playlistURI }}
+              style={styles.image}
+              resizeMode="cover"
+            />
+            <Text style={styles.description} numberOfLines={4}>
+              {data.length ? data[0].description : ''}
+            </Text>
+            <Subtitle style={styles.subtitle}>IN THIS SERIES</Subtitle>
+            {/* {
```

Comment:
💁  **suggstion**
if we don't need this, we can probably just delete it. we can always see git history if we need it back again.